crates/kitsune2/src/lib.rs

```diff
@@ -37,3 +36,215 @@ pub fn default_builder() -> Builder {
         local_agent_store: factories::CoreLocalAgentStoreFactory::create(),
     }
 }
+
+#[cfg(test)]
+mod test {
+    use crate::default_builder;
+    use bytes::Bytes;
+    use kitsune2_api::{
+        BoxFut, DhtArc, DynSpace, DynSpaceHandler, K2Result, KitsuneHandler,
+        LocalAgent, OpId, SpaceHandler, SpaceId, Timestamp,
+    };
+    use kitsune2_core::{
+        factories::{
+            config::{CoreBootstrapConfig, CoreBootstrapModConfig},
+            MemOpStoreFactory, MemoryOp,
+        },
+        Ed25519LocalAgent,
+    };
+    use kitsune2_gossip::{K2GossipConfig, K2GossipModConfig};
+    use kitsune2_test_utils::{
+        bootstrap::TestBootstrapSrv, enable_tracing, iter_check, random_bytes,
+        space::TEST_SPACE_ID,
+    };
+    use kitsune2_transport_tx5::config::{
+        Tx5TransportConfig, Tx5TransportModConfig,
+    };
+    use sbd_server::SbdServer;
+    use std::sync::Arc;
+
+    fn create_op_list(num_ops: u16) -> (Vec<Bytes>, Vec<OpId>) {
+        let mut ops = Vec::new();
+        let mut op_ids = Vec::new();
+        for _ in 0..num_ops {
+            let op =
+                MemoryOp::new(Timestamp::from_micros(0), random_bytes(256));
+            let op_id = op.compute_op_id();
+            ops.push(op.into());
+            op_ids.push(op_id);
+        }
+        (ops, op_ids)
+    }
+
+    #[derive(Debug)]
+    struct HolochainKitsuneHandler;
+    impl KitsuneHandler for HolochainKitsuneHandler {
+        fn create_space(
+            &self,
+            _space_id: SpaceId,
+        ) -> BoxFut<'_, K2Result<DynSpaceHandler>> {
+            Box::pin(async {
+                let space_handler: DynSpaceHandler =
+                    Arc::new(HolochainSpaceHandler);
+                Ok(space_handler)
+            })
+        }
+    }
+
+    #[derive(Debug)]
+    struct HolochainSpaceHandler;
+    impl SpaceHandler for HolochainSpaceHandler {}
+
+    async fn make_kitsune_space(
+        signal_server_url: &str,
+        bootstrap_server_url: &str,
+    ) -> DynSpace {
+        let mut kitsune_builder =
+            default_builder().with_default_config().unwrap();
+        kitsune_builder
+            .config
+            .set_module_config(&CoreBootstrapModConfig {
+                core_bootstrap: CoreBootstrapConfig {
+                    server_url: bootstrap_server_url.to_owned(),
+                    backoff_max_ms: 1000,
+                    ..Default::default()
+                },
+            })
+            .unwrap();
+        kitsune_builder.op_store = MemOpStoreFactory::create();
```

Comment:
Don't replace a factory after you've already set the default config... you end up with a config for the wrong factory.

```rust
let mut kitsune_builder = Builder {
    op_store: MemOpStoreFactory::create(),
    ..default_builder()
}.with_default_config().unwrap();
```

Reply:
Also, the MemOpStoreFactory *is* the default, so you don't need to do this at all.